src/lastdose.cpp

```diff
@@ -60,13 +57,14 @@ Rcpp::List lastdose_impl(Rcpp::NumericVector id,
                          Rcpp::LogicalVector comment) {
 
   amt = Rcpp::clone(amt);
-  bool use_comp1 = sort1[0];
+  bool obs_first = sort1[0];
   bool use_fill = !back_calc[0];
   std::vector<double> idn;
   std::vector<int> idstart;
   std::vector<int> idend;
   double lastid = -1E9;
   int nrows = id.size();
+  const int max_addl_pos = nrows + 1;
```

Comment:
When `obs_first` is `TRUE`, then any record from additional doses will always get pushed toward the bottom of the data set at a given time.